### PR TITLE
feat: Allow user to register its own hashers

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use blockstore::InMemoryBlockstore;
+    /// BitswapBehaviour::builder(InMemoryBlockstore::new())
+    ///     .protocol_prefix("/celestia/celestia")
+    ///     .build()?;
+    /// ```
+    pub fn protocol_prefix(&mut self, prefix: &str) {
```
Comment (zvolin) [inline src/builder.rs:49]: shouldn't those take and return the builder to use it like in doc?

Comment (oblique) [reply src/builder.rs]: yeah i forgot to do it

**Review by fl0rek — APPROVED** [review]
🚀

File: src/builder.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    /// Client will set `send_dont_have` flag on each query.
+    pub fn client_set_send_dont_have(mut self, enable: bool) -> Self {
+        self.client.set_send_dont_have = enable;
+        self
+    }
+
+    /// Register extra [`Multihasher`].
+    ///
+    /// Every registration overrides the previous onces. By default `BitswapBehaviourBuilder`
+    /// is pre-loaded with [`StandardMultihasher`].
```
Comment (zvolin) [inline src/builder.rs:78]:
```suggestion
    /// Every registration adds new hasher to `BitswapBehavior`. Hashers are used to reconstruct the `Cid` from the received data.
    /// `BitswapBehavior` will try them in the reverse order they were registered until some successfully constructs `Multihash`.
    /// By default `BitswapBehaviourBuilder` is pre-loaded with [`StandardMultihasher`].

```

**Review by zvolin — COMMENTED** [review]
looks good

**Review by zvolin — APPROVED** [review]

**Pull request merged**